tests/workflows/workflow_test_helpers.py

```
@@ -43,6 +43,50 @@
 
 
+def _cargo_metadata() -> dict[str, object]:
+    """Return parsed workspace metadata from `cargo metadata --no-deps`.
+
+    Returns
+    -------
+    dict[str, object]
+        Parsed metadata mapping emitted by Cargo.
+    """
+    completed = subprocess.run(  # noqa: S603,S607  # FIXME: uses trusted test-only PATH-resolved tool
+        ["cargo", "metadata", "--format-version", "1", "--no-deps"],
```

Ruff (0.15.9)

[error] Starting a process with a partial executable path

(S607)